war/pom.xml

```diff
@@ -538,36 +571,67 @@ THE SOFTWARE.
         </plugins>
       </build>
     </profile>
+
+      
+    <!--
+        The following profiles are required to integration the node/gulp build into this maven build.
+        Hopefully we can push these profiles down into a parent pom.
+        See https://github.com/tfennelly/jenkins-js-builder#maven-integration
+    -->
+      
+    <!--
+        Manually download node and NPM for the frontend plugin so it gets cached in the local maven repo.
+        Otherwise frontend will be downloading it all the time on the CI servers. The download-maven-plugin 
+        caches it's downloads in the local maven repo.
+    -->
+    <profile>
+        <id>node-classifier-linux</id>
+        <activation><os><family>Linux</family><arch>amd64</arch></os></activation>
+        <properties><node.download.file>node-v${node.version}-linux-x64.tar.gz</node.download.file></properties>
+    </profile>
     <profile>
-      <!-- profile for running Grunt and optimize static resources -->
-      <id>grunt</id>
+        <id>node-classifier-mac</id>
+        <activation><os><family>mac</family></os></activation>
+        <properties><node.download.file>node-v${node.version}-darwin-x64.tar.gz</node.download.file></properties>
+    </profile>
+    <profile>
+        <id>node-classifier-windows</id>
+        <activation><os><family>windows</family></os></activation>
+        <properties><node.download.file>win-x64/node.exe</node.download.file></properties>
+    </profile>
+    <profile>
+        <id>node-download</id>
+        <activation><file><exists>package.json</exists></file></activation>
         <build>
             <plugins>
                 <plugin>
-                    <groupId>org.codehaus.mojo</groupId>
-                    <artifactId>exec-maven-plugin</artifactId>
+                    <groupId>com.googlecode.maven-download-plugin</groupId>
+                    <artifactId>download-maven-plugin</artifactId>
+                    <version>1.2.1</version>
                     <executions>
                         <execution>
-                            <id>install</id>
-                            <phase>validate</phase>
+                            <id>get-node</id>
+                            <phase>initialize</phase>
                             <goals>
-                                <goal>exec</goal>
+                                <goal>wget</goal>
```

Comment:
Node download exists in `frontend-maven-plugin` why another one plugin is required?